contribute/engineering/backend/instrumentation.md

```diff
@@ -0,0 +1,207 @@
+
+If label values originates from user input they **should** be validated. Use `metricutil.SanitizeLabelName(<label value>`) from _pkg/infra/metrics/metricutil_ package to sanitize label names. Very **important** to only allow a pre-defined set of labels to minimize the risk of high cardinality problems.
+
+Be **careful** to not expose any sensitive information in label values, e.g. secrets, credentials etc.
+
+### Guarantee the existence of metrics
+
+If you want to guarantee the existence of metrics before any observations has happened there's a couple of helper methods available in the _pkg/infra/metrics/metricutil_ package.
+
+### How to collect and visualize metrics locally
+
+1. Start Prometheus
+
+   ```bash
+   make devenv sources=prometheus
+   ```
+
+2. Use Grafana Explore or dashboards to query any exported Grafana metrics
+
+## Traces
+
+A distributed trace is data that tracks an application request as it flows through the various parts of an application. The trace records how long it takes each application component to process the request and pass the result to the next component. Traces can also identify which parts of the application trigger an error.
+
+### Naming conventions
+
+Span names should follow the [guidelines from OpenTelemetry](https://opentelemetry.io/docs/reference/specification/trace/api/#span).
+
+| Span Name               | Guidance                                                 |
+| ----------------------- | -------------------------------------------------------- |
+| get                     | Too general                                              |
+| get_account/42          | Too specific                                             |
+| get_account             | Good, and account_id=42 would make a nice Span attribute |
+| get_account/{accountId} | Also good (using the “HTTP route”)                       |
+
+Span attribute and span event attributes should follow the [Attribute naming specification from OpenTelemetry](https://opentelemetry.io/docs/reference/specification/common/attribute-naming/). Good attribute key examples:
+
+- service.version
+- http.status_code
+
+See [Trace semantic conventions from OpenTelemetry](https://opentelemetry.io/docs/reference/specification/trace/semantic_conventions/) for additional conventions regarding well-known protocols and operations.
+
+### Span names and high cardinality
+
+Be careful with what span names you add/accept. Using/allowing too many span names could result in high cardinality problems.
+
+### Validate and sanitize input coming from user input
+
+If span names, attribute or event values originates from user input they **should** be validated and sanitized. It's very **important** to only allow a pre-defined set of span names to minimize the risk of high cardinality problems.
```

Comment:
I am curious about attribute value sanitization - do we have any util or that? I see you have mention one for label names in a section above

Reply:
I don't think we have one for this purpose and not sure the one for labels work as expected. Could also be that opentelemetry validates this internally :shrug: